### DISCUSSION TITLE
How to update GIT and cURL old versions to support homebrew installation on REDHAT 7

### DISCUSSION THREAD

**Discussion by shivak3** [Homebrew/discussions#3393] [Linux] [answered]
I have Linux Server with GIT version 1.18.1 and CURL version 7.29.0 loaded from rpm packages.

How to install latest versions to match with homebrew installation? Please help me.

**Comment by iMichka** [comment, marked as answer]
You can find some inspiration on how to download, compile these by looking at this pull request: https://github.com/Homebrew/brew/pull/13231

We plan to automatise the installation of newer git/curl if necessary, we are working on this, but it may take some time before it is automatised.

**Reply by shivak3** [reply]
Hi iMichka,

Thank you for reply.  Below scripts worked for me to build and install Curl and git versions and finally helped me to install homebrew package.

~~~
RUN curl -sL https://curl.se/download/curl-7.82.0.tar.bz2 | tar xj \
    && cd /curl-7.82.0 \
    && make configure \
    && ./configure --prefix=/usr/local --with-openssl \
    && make \
    && make install \
    && cd / \
    && rm -rf /curl-7.82.0

RUN curl -sL http://mirrors.edge.kernel.org/pub/software/scm/git/git-2.28.0.tar.gz | tar xz \
    && cd /git-2.28.0 \
    && make configure \
    && ./configure --prefix=/usr/local --with-curl=/usr/local \
    && make install NO_TCLTK=1 \
    && cd / \
    && rm -rf /git-2.28.0


RUN echo "export HOMEBREW_GIT_PATH=/usr/local/bin/git" >> /home/linuxbrew/.bashrc
RUN echo "export HOMEBREW_CURL_PATH=/usr/local/bin/curl" >> /home/linuxbrew/.bashrc
RUN echo "export HOMEBREW_FORCE_BREWED_CA_CERTIFICATES=1" >> /home/linuxbrew/.bashrc

~~~
